x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_domains_logic.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+import { DEFAULT_META } from '../../../shared/constants';
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { updateMetaPageIndex } from '../../../shared/table_pagination';
+import { EngineLogic } from '../engine';
+
+import {
+  CrawlerData,
+  CrawlerDataFromServer,
+  CrawlerDomain,
+  CrawlerDomainFromServer,
+} from './types';
+import { crawlerDataServerToClient, crawlerDomainServerToClient } from './utils';
+
+export interface CrawlerDomainsValues {
+  dataLoading: boolean;
+  domains: CrawlerDomain[];
+  meta: Meta;
+}
+
+interface CrawlerDomainsResponse {
+  results: CrawlerDomainFromServer[];
+  meta: Meta;
+}
+
+interface CrawlerDomainsActions {
+  deleteDomain(domain: CrawlerDomain): { domain: CrawlerDomain };
+  fetchCrawlerDomainsData(): void;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+  onReceiveData(domains: CrawlerDomain[], meta: Meta): { domains: CrawlerDomain[]; meta: Meta };
+  crawlerDomainDeleted(data: CrawlerData): { data: CrawlerData };
+}
+
+export const CrawlerDomainsLogic = kea<MakeLogicType<CrawlerDomainsValues, CrawlerDomainsActions>>({
```

Comment:
When we build Logic files, I think they can be 1 of 2 types:

1. Resource based - Representing a discrete piece of data. Usually associated with CRUD operations.
2. View based - Represents the actions and state that a user can take on a particular view.

Some times we mix the two. Sometimes we keep these separated.

I chose to switch from `CrawlerOverviewLogic` (a view based logic) to `CrawlerDomainsLogic` (a resource based logic). Particularly now because we have an API to specifically list `CrawlerDomain`s. The idea is that it should encapsulate all resource CRUD operations, but NOT be concerned with any view details. It can publish events (i.e., actions) that view resources can listen for.